File: app/src/main/java/com/sudoajay/pdf_viewer/SharedPreference/PrefManager.java
package com.sudoajay.pdf_viewer.SharedPreference;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;

import com.sudoajay.pdf_viewer.R;

public class PrefManager {
    private SharedPreferences pref;
    private SharedPreferences.Editor editor;
    private Context _context;


    @SuppressLint("CommitPrefEdits")
    public PrefManager(Context context) {
        this._context = context;
        // shared pref mode
        int PRIVATE_MODE = 0;
        pref = _context.getSharedPreferences(context.getString(R.string.MY_PREFS_NAME), PRIVATE_MODE);
        editor = pref.edit();
    }

    public void setFirstTimeLaunch(boolean isFirstTime) {
        editor.putBoolean(_context.getString(R.string.isFirstTimeLaunch), isFirstTime);
        editor.apply();
    }

    public boolean isFirstTimeLaunch() {
        return pref.getBoolean(_context.getString(R.string.isFirstTimeLaunch), true);
    }

}

File: app/src/main/java/com/sudoajay/pdf_viewer/sharedPreference/PrefManager.kt
package com.sudoajay.pdf_viewer.sharedPreference

import android.annotation.SuppressLint
import android.content.Context
import android.content.SharedPreferences
import android.content.SharedPreferences.Editor
import com.sudoajay.pdf_viewer.R

class PrefManager @SuppressLint("CommitPrefEdits") constructor(private val _context: Context) {
    private val pref: SharedPreferences
    private val editor: Editor

    var isFirstTimeLaunch: Boolean
        get() = pref.getBoolean(_context.getString(R.string.isFirstTimeLaunch), true)
        set(isFirstTime) {
            editor.putBoolean(_context.getString(R.string.isFirstTimeLaunch), isFirstTime)
            editor.apply()
        }

    init {
        // shared pref mode
        val privateMode = 0
        pref = _context.getSharedPreferences(_context.getString(R.string.MY_PREFS_NAME), privateMode)
        editor = pref.edit()
    }
}